File: clients/scala/src/test/scala/com/schmitech/orbit/ApiClientTest.scala
package com.schmitech.orbit

import org.junit.Test
import org.junit.Assert._
import org.junit.Assume._

class ApiClientTest {
  @Test def integrationNonStreaming(): Unit = {
    assumeTrue("Set ORBIT_INTEGRATION=1 to enable", System.getenv("ORBIT_INTEGRATION") == "1")
    val url = Option(System.getenv("ORBIT_URL")).getOrElse("http://localhost:3000")
    val c = new ApiClient(url)
    var all = new StringBuilder
    c.streamChat("ping", false) { r => all ++= r.text }
    assertTrue(all.length >= 0) // smoke
  }
}

File: clients/kotlin/bin/test/com/schmitech/orbit/ApiClientTest.kt
package com.schmitech.orbit

import org.junit.jupiter.api.Assertions.assertFalse
import org.junit.jupiter.api.Assumptions.assumeTrue
import org.junit.jupiter.api.Test

class ApiClientTest {
    @Test
    fun integrationNonStreaming() {
        assumeTrue(System.getenv("ORBIT_INTEGRATION") == "1", "Set ORBIT_INTEGRATION=1 to enable")
        val url = System.getenv("ORBIT_URL") ?: "http://localhost:3000"
        val client = ApiClient(url)
        var all = StringBuilder()
        client.streamChat("ping", false) { chunk -> all.append(chunk.text) }
        Thread.sleep(1000) // allow async callback to complete in simple demo
        assertFalse(all.isEmpty())
    }
}

